src/lib.rs

```diff
@@ -331,6 +349,10 @@ mod tests {
     #[test]
     fn test_eq() {
         let c1 = Cactus::new().child(1).child(2);
+        assert_eq!(c1, c1);
+        let z = c1.child(4);
```

Comment:
nitpick, but if we have `c1` and `c2`, shouldn't we also have `z1` and `z2`, not `z` and `z2`?

Reply:
Did we want to have a test which finds pointer equality a little later in the chain? i.e. testing a few iterations of the loop?

Reply:
Oops, the `z` was me messing around with something else -- I forgot to go back and make this consistent. Fixed now.

Reply:
This test is designed not to find pointer equality at the first level: it finds it at the second level. That is morally equivalent, IMHO, to all levels beyond that.